src/components/crowdsale/utils.js

```diff
@@ -284,6 +284,16 @@ export const isFinalized = async ({ methods }, crowdsaleExecID) => {
   return is_finalized
 }
 
+export const isEnded = async ({ methods }, crowdsaleExecID) => {
```

Comment:
There is only end by time is checked here, but crowdsale can be ended also if it is full (hardcap is sold out). But I think it could be made in a separate PR.

Reply:
Added function to check if a crowdsale is sold out, and validate the button